File: lista/JAVA/questao3/ContaBancaria.java
package questao3;
/*
Implemente uma classe ContaBancaria com atributos saldo, titular e
métodos depositar e sacar. Use encapsulamento para proteger o atributo saldo
 */

public class ContaBancaria {
    String titular;
    private double saldo;

    public ContaBancaria(String titular, double saldo){
        this.titular = titular;
        this.saldo = saldo;
    }

    public double getSaldo() {
        return saldo;
    }
    public void setSaldo(double saldo) {
        this.saldo = saldo;
    }

    public void exibirConta(){
        System.out.println("Titular: " + titular);
        System.out.println("Saldo: " + getSaldo());
    }

    public void depositar(double valor){
        if (valor > 0){
            System.out.println("Depositando R$" + valor);
            saldo += valor;
        }
        else {
            System.out.println("Valor inválido");
        }
    }
    public void sacar(double valor){
        if(saldo >= valor && valor > 0){
            System.out.println("Sacando R$" + valor);
            saldo -= valor;
        }
        else {
            System.out.println("Saldo insuficiente ou valor inválido\n");
        }
    }
    public void exibirSaldo(){
        System.out.println("Saldo Atual: " + saldo + "\n");
    }

    public static void main(String[] args) {
        ContaBancaria c1 = new ContaBancaria("João", 1000);
        c1.exibirConta();

        c1.exibirSaldo();
        c1.depositar(500);
        c1.exibirSaldo();
        c1.sacar(300);
        c1.exibirSaldo();

        c1.exibirConta();
    }

}

File: lista/GO/questao3/contaBancaria.go
package main

import (
	"fmt"
)

type ContaBancaria struct {
	Titular string
	saldo   float64
}

func NovaContaBancaria(titular string, saldoInicial float64) ContaBancaria {
	return ContaBancaria{Titular: titular, saldo: saldoInicial}
}

func (c *ContaBancaria) GetSaldo() float64 {
	return c.saldo
}

func (c *ContaBancaria) SetSaldo(novoSaldo float64) {
	c.saldo = novoSaldo
}

func (c ContaBancaria) ExibirConta() {
	fmt.Printf("Titular: %s\n", c.Titular)
	fmt.Printf("Saldo: R$%.2f\n", c.GetSaldo())
}

func (c *ContaBancaria) Depositar(valor float64) {
	if valor > 0 {
		fmt.Printf("Depositando R$%.2f\n", valor)
		c.saldo += valor
	} else {
		fmt.Println("Valor inválido")
	}
}

func (c *ContaBancaria) Sacar(valor float64) {
	if c.saldo >= valor && valor > 0 {
		fmt.Printf("Sacando R$%.2f\n", valor)
		c.saldo -= valor
	} else {
		fmt.Println("Saldo insuficiente ou valor inválido\n")
	}
}

func (c ContaBancaria) ExibirSaldo() {
	fmt.Printf("Saldo Atual: R$%.2f\n\n", c.saldo)
}

func main() {
	c1 := NovaContaBancaria("João", 1000)

	c1.ExibirConta()

	c1.ExibirSaldo()
	c1.Depositar(500)
	c1.ExibirSaldo()
	c1.Sacar(300)
	c1.ExibirSaldo()

	c1.ExibirConta()
}
